kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4

```diff
@@ -332,7 +332,7 @@ qualifiedName
         ( '.'
             {helper.recoverScope( name ); count++;}
             n2=nameRef
-            {name=getOriginalText( $n2.ctx );}
+            {name=getOriginalText( $n2.ctx );  qn.add( name ); helper.validateVariable( $n1.ctx, qn, name ); }
```

Comment:
It was missing the actual check the qualifiedName composed so far is valid (will check against the last chunk `name` the `qn` I understand is composed for reporting error only).